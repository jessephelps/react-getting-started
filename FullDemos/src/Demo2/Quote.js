import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './Quote.css';

class Quote extends Component {
  render() {
    return (
      <div className="Quote">
        {this.props.text}
      </div>
    );
  }
}

export default Quote;

Quote.propTypes = {
  text: PropTypes.string,
};
