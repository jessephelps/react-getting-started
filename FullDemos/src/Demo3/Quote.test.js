import React from 'react';
import renderer from 'react-test-renderer';
import Quote from './Quote';

it('renders a div with some text', () => {
  // arrange
  const testText = 'This is some test text.';
  
  // act
  const compoent = renderer.create(<Quote text={testText} />);
  const tree = compoent.toJSON();

  // assert
  expect(tree).toMatchSnapshot();
  expect(tree.children[0]).toBe(testText);
});
