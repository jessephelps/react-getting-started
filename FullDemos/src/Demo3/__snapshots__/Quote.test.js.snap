// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a div with some text 1`] = `
<div
  className="Quote"
>
  This is some test text.
</div>
`;
